src/test/java/chess/domain/chessboard/ChessBoardTest.java

```diff
@@ -0,0 +1,186 @@
+        var sut = new ChessBoard(
+                new HashMap<>() {{
+                    put(new Position(File.f, Rank.ONE), new King(Team.WHITE));
+                }}
+        );
+
+        var result = sut.move(source, target, WHITE_TURN);
+
+        assertThat(result).isEqualTo(BLACK_TURN);
+    }
+
+    @Test
+    @DisplayName("왕을 잡으면 게임이 종료된다.")
+    void ChessBoard_Game_over_when_king_is_out() {
+        Position source = new Position(File.a, Rank.SEVEN);
+        Position target = new Position(File.b, Rank.EIGHT);
+        var sut = new ChessBoard(
+                new HashMap<Position, Piece>() {{
+                    put(new Position(File.b, Rank.EIGHT), new King(BLACK));
+                    put(new Position(File.a, Rank.SEVEN), new WhitePawn());
+                }}
+        );
+
+        var result = sut.move(source, target, WHITE_TURN);
+
+        assertThat(result).isEqualTo(GAME_OVER);
+    }
+
+    @Test
+    @DisplayName("비어있는 공간을 움직이려고 하면 예외가 발생한다.")
+    void ChessBoard_Throw_exception_when_move_empty() {
+        Position source = new Position(File.a, Rank.SEVEN);
+        Position target = new Position(File.b, Rank.EIGHT);
+        var sut = new ChessBoard(
+                new HashMap<>() {{
+                }}
+        );
+
+        assertThatThrownBy(() -> {
+            sut.move(source, target, WHITE_TURN);
+        }).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    /*
+    R.......
+    ........
+    ........
+    p.......
+    ........
+    ........
+    ........
+    ........
+    */
+    @Test
+    @DisplayName("이동 경로에 방해물이 있는 경우 예외가 발생한다.")
+    void ChessBoard_Throw_exception_route_has_obstacle() {
+        Position source = new Position(File.a, Rank.EIGHT);
+        Position target = new Position(File.a, Rank.ONE);
+        var sut = new ChessBoard(
+                new HashMap<>() {{
+                    put(new Position(File.a, Rank.EIGHT), new Rook(BLACK));
+                    put(new Position(File.a, Rank.FIVE), new WhitePawn());
+                }}
+        );
+
+        assertThatThrownBy(() -> {
+            sut.move(source, target, BLACK_TURN);
+        }).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    /*
+    R.......
+    ........
+    ........
+    P.......
+    ........
+    ........
+    ........
+    ........
+     */
+    @Test
+    @DisplayName("목적지에 아군이 있는 경우 예외가 발생한다.")
+    void ChessBoard_Throw_exception_target_is_team() {
+        Position source = new Position(File.a, Rank.EIGHT);
+        Position target = new Position(File.a, Rank.FIVE);
+        var sut = new ChessBoard(
+                new HashMap<>() {{
+                    put(new Position(File.a, Rank.EIGHT), new Rook(BLACK));
+                    put(new Position(File.a, Rank.FIVE), new BlackPawn());
+                }}
+        );
+
+        assertThatThrownBy(() -> {
+            sut.move(source, target, BLACK_TURN);
+        }).isInstanceOf(IllegalArgumentException.class);
```

Comment:
```suggestion
        assertThatThrownBy(() -> sut.move(source, target, BLACK_TURN))
                .isInstanceOf(IllegalArgumentException.class);
```
람다식은 하나의 명령줄만 있는 경우 {}를 사용하지 않아도됩니다.